Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Staging/SetpointController.mo

```diff
@@ -0,0 +1,548 @@
+    final nSta=nSta) "Design and minimum capacities for relevant chiller stages"
+    annotation (Placement(transformation(extent={{-270,-180},{-250,-160}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Staging.Subsequences.PartLoadRatios PLRs(
+    final anyVsdCen=anyVsdCen,
+    final nSta=nSta,
+    final posDisMult=posDisMult,
+    final conSpeCenMult=conSpeCenMult,
+    final anyOutOfScoMult=anyOutOfScoMult,
+    final varSpeStaMin=varSpeStaMin,
+    final varSpeStaMax=varSpeStaMax) "Operative and staging part load ratios"
+    annotation (Placement(transformation(extent={{-182,-200},{-162,-180}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Staging.Subsequences.Up staUp(
+    final have_WSE=have_WSE,
+    final serChi=serChi,
+    final effConTruDelay=effConTruDelay,
+    final faiSafTruDelay=faiSafTruDelay,
+    final shortTDelay=shortTDelay,
+    final longTDelay=longTDelay,
+    final faiSafTDif=faiSafTDif,
+    final TDifHys=TDifHys,
+    final smallTDif=smallTDif,
+    final largeTDif=largeTDif,
+    final faiSafDpDif=faiSafDpDif,
+    final dpDifHys=dpDifHys,
+    final effConSigDif=effConSigDif) "Stage up conditions"
+    annotation (Placement(transformation(extent={{-100,-120},{-80,-100}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Staging.Subsequences.Down staDow(
+    final have_WSE=have_WSE,
+    final serChi=serChi,
+    final parLoaRatDelay=parLoaRatDelay,
+    final faiSafTruDelay=faiSafTruDelay,
+    final faiSafTDif=faiSafTDif,
+    final faiSafDpDif=faiSafDpDif,
+    final TDif=TDif,
+    final TDifHys=TDifHys) "Stage down conditions"
+    annotation (Placement(transformation(extent={{-100,-240},{-80,-220}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uPla "Plant enable signal"
+                          annotation (Placement(
+        transformation(extent={{-440,-100},{-400,-60}}),  iconTransformation(
+          extent={{-140,-230},{-100,-190}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput uIni(final min=0, final max=nSta)
+    "Initial chiller stage (at plant enable)" annotation (Placement(
+        transformation(extent={{-440,-30},{-400,10}}),   iconTransformation(
+          extent={{-140,-148},{-100,-108}})));
+
+  CDL.Interfaces.BooleanOutput y "Chiller stage change edge signal" annotation (
+     Placement(transformation(extent={{120,-200},{160,-160}}),
+        iconTransformation(extent={{100,-90},{140,-50}})));
+  CDL.Interfaces.BooleanInput chaPro "Stage change process status signal"
+    annotation (Placement(transformation(extent={{-440,-140},{-400,-100}}),
+        iconTransformation(extent={{-140,-190},{-100,-150}})));
+  Subsequences.Change cha(
+    final nSta=nSta,
+    final delayStaCha=delayStaCha) "Stage change assignment"
+    annotation (Placement(transformation(extent={{-20,-180},{0,-160}})));
+  Subsequences.ChillerIndices chiInd(
```

Comment:
add comment. Also, it would be better to modify the text to be more organized.

Reply:
ups, it looks like I did not sort through some of these, will do, thanks!